### PR TITLE
[WIP] Converting to Swift 3

### PR DESCRIPTION
**Pull request by floskel**
I have started the process of converting to Swift 3. 
This requires `RxSwift`/`RxCocoa` to be stable, which won't happen for some weeks. 

I have converted the Action project and made it build using the `RxSwift` develop branch. When proper beta/rc are available we should update to those. 

As of 9th of September, the Demo project is not compiling, as the Podfile cannot use a specific branch. 

I will keep updating this PR as `RxSwift` releases new candidates.  
Feel free to contribute or comment.

### CODE REVIEW

**Comment by ashfurrow** [comment]
:100:

File: Action.swift [hunk 1]
```diff
@@ -23,83 +23,75 @@ public final class Action<Input, Element> {
...
-    private let disposeBag = DisposeBag()
+    fileprivate let executingQueue = DispatchQueue(label: "com.ashfurrow.Action.executingQueue", attributes: [])
+    fileprivate let disposeBag = DisposeBag()
 
-    public init<B: BooleanType>(enabledIf: Observable<B>, workFactory: WorkFactory) {
-        self._enabledIf = enabledIf.map { booleanType in
-            return booleanType.boolValue
+    public init<B: ExpressibleByBooleanLiteral>(enabledIf: Observable<B> = Observable.just(true), workFactory: @escaping WorkFactory) {
+        self._enabledIf = enabledIf.map { (booleanLiteral) -> Bool in
+            return booleanLiteral as! Bool
```
Comment (ishkawa) [inline Action.swift]: [`BooleanType` has been renamed to `Bool`](https://github.com/apple/swift/blob/4a24b069e0388cdac02551cda06a0d0ef1298436/test/stdlib/Renames.swift#L124).
We can make this initializer non-generic and replace the type parameter with `Bool`.

File: Action.swift [hunk 2]
```diff
@@ -23,83 +23,75 @@ public final class Action<Input, Element> {
...
         }.bindTo(_enabled).addDisposableTo(disposeBag)
 
-        self.inputs
-            .subscribeNext { [weak self] input in
-                self?._execute(input)
-            }
-            .addDisposableTo(disposeBag)
+        self.inputs.subscribe(onNext: { [weak self] (input) in
+            self?._execute(input)
+        }, onError: nil, onCompleted: nil, onDisposed: nil).addDisposableTo(disposeBag)
```
Comment (ishkawa) [inline Action.swift]: nit: `, onError: nil, onCompleted: nil, onDisposed: nil` can be omitted.

File: Action.swift [hunk 3]
```diff
@@ -134,9 +127,9 @@ public extension Action {
 
         // Make sure we started executing and we're accidentally disabled.
         guard startedExecuting else {
-            let error = ActionError.NotEnabled
+            let error = ActionError.notEnabled
             self._errors.onNext(error)
-            buffer.onError(error)
+            buffer.onError(error as Error)
```
Comment (ishkawa) [inline Action.swift]: nit: `as Error` can be omitted.

**Comment by floskel** [comment]
Thanks @ishkawa. Added a few fixes for your comments.

**Comment by ashfurrow** [comment]
Thanks a lot for contributing! I've invited you to join the RxSwiftCommunity GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/RxSwiftCommunity/contributors) and feel free to reach out with any questions.

**Comment by floskel** [comment]
Thanks 👍

**Pull request merged**